tests/pkg/webhook/pods.go

```diff
@@ -156,14 +204,27 @@ func admitPods(ar v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
 			return &reviewResponse
 		}
 
-		if storeInfo.Status.LeaderCount > 0 && tc.Status.TiKV.StatefulSet.Replicas > 1 {
+		beforeCount := kvLeaders[name]
+		afterCount := storeInfo.Status.LeaderCount
+		glog.Infof("before evict the leader is %d after evict the leader is %d", beforeCount, afterCount)
+
+		if beforeCount != 0 && afterCount >= beforeCount && tc.Status.TiKV.StatefulSet.Replicas > 1 {
 			time.Sleep(10 * time.Second)
 			glog.Errorf("kv leader is not zero, can't be deleted namespace %s name %s leaderCount %d", nameSpace, name, storeInfo.Status.LeaderCount)
 			os.Exit(3)
 		} else {
 			glog.Infof("savely delete pod namespace %s name %s", nameSpace, name)
 		}
+
+		kvShoot[shootName] = true
+
+	}
+
+	if kvLeaders, err = getAllKVLeaders(tc, httpClient); err != nil {
```

Comment:
the `getAllKVLeaders` should be called outside admitPods, best in the init phase of the webhook server.